h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -718,6 +728,58 @@ public final double likelihood(double yr, double ym){
         return .5 * deviance(yr,ym);
     }
 
+    // more time-consuming likelihood calculation used for the final scoring
+    public final double likelihood(double w, double yr, double[] ym) {
+      double prediction = ym[0];
+      double probabilityOf1;
+      switch (_family) {
+        case gaussian:
+          return -.5 * (w * Math.pow(yr - prediction , 2) / _dispersion_estimated 
+                  + log(_dispersion_estimated / w) + LOG2PI);
+        case binomial:
+          // if probability is not given, then it is 1.0 if 1 is predicted and 0.0 if 0 is predicted
+          probabilityOf1 = ym.length > 1 ? ym[2] : ym[0]; // probability of 1 equals prediction
+          return w * (yr * log(probabilityOf1) + (1-yr) * log(1 - probabilityOf1)) 
+                  + w * (Gamma.digamma(2) - Gamma.digamma(yr + 1) 
+                  - Gamma.digamma(1 - yr + 1));
+        case quasibinomial:
+          // if probability is not given, then it is 1.0 if 1 is predicted and 0.0 if 0 is predicted
+          probabilityOf1 = ym.length > 1 ? ym[2] : ym[0]; // probability of 1 equals prediction
+          if (yr == prediction)
+            return 0;
+          else if (prediction > 1) // check what are possible values?
+            return -w * (yr * log(probabilityOf1));
+          else
+            return -w * (yr * log(probabilityOf1) + (1 - yr) * log(1 - probabilityOf1));
+        case fractionalbinomial:
+          // if probability is not given, then it is 1.0 if 1 is predicted and 0.0 if 0 is predicted
+          probabilityOf1 = ym.length > 1 ? ym[2] : ym[0]; // probability of 1 equals prediction
+          if (yr == prediction)
+            return 0;
+          return w * ((MathUtils.y_log_y(yr, probabilityOf1)) + MathUtils.y_log_y(1 - yr, 1 - probabilityOf1));
+        case poisson:
+          return w * (yr * log(prediction) - prediction - Gamma.digamma(yr + 1)); // gamma(n) = (n-1)!
+        case negativebinomial:
+          return yr * log(_invTheta * prediction / w) - (yr + w/_invTheta) * log(1 + _invTheta * prediction / w) 
+                  + log(Gamma.gamma(yr + w/_invTheta) / (Gamma.gamma(yr + 1) * Gamma.gamma(w/_invTheta)));
```

Comment:
The _invTheta here is the dispersion parameter.  Please enable the dispersion parameter calculation here as well and use that.